### PR TITLE
feat: add account limits screen

### PR DESCRIPTION
**Pull request by Extheoisah**
**Description**

This PR adds account limits screen to users settings in the app and should be merged after the corresponding [account limits PR](https://github.com/GaloyMoney/galoy/pull/1953) from the Galoy backend repo has been merged.

To test the PR, 
- run the Galoy backend locally and switch to [this](https://github.com/GaloyMoney/galoy/pull/1953) branch
- next, checkout to this commit [`65c649a`](https://github.com/GaloyMoney/galoy/commit/65c649a8ce617f06b4765bed77a4680eeb2accb7) and start the main API. 
- next, in the debugging section of the mobile app, change your backend development server to local and connect your android/iOS simulator/device to the local server (for android run `yarn adb` to sync ports)
- You should find the limits screen under `Settings -> Accounts -> Account Limits`

Update:
- updated the screen to not differentiate USD and BTC transactions.
Screenshot 2022-12-20 at 12 53 15
Screenshot 2022-12-18 at 23 04 56

### CODE REVIEW

**Comment by daviroo** [comment]
Reviewing now.  Here are some screenshots for others to look at.

image

image

File: app/hooks/use-account-limits.ts
Comment (daviroo) [inline app/hooks/use-account-limits.ts:76]: Not sure what the purpose of this is?

File: app/screens/settings-screen/account-screen.tsx
Comment (daviroo) [inline app/screens/settings-screen/account-screen.tsx:65]: What about some type of exclamation icon?  e.g. `exclamationcircleo`.  You can see the choices here https://oblador.github.io/react-native-vector-icons/.

**Review by daviroo — CHANGES_REQUESTED** [review]
Just a few comments.  There is a bit of margin at the top which seems out of place.  I've included a screenshot to show what I mean.

image

File: app/hooks/use-account-limits.ts
Comment (Extheoisah) [reply app/hooks/use-account-limits.ts:76]: > Not sure what the purpose of this is?

This persists the limits data and checks for changes in the user account limits. So this ensures real-time change as opposed to using a if statement

File: app/hooks/use-account-limits.ts
Comment (Extheoisah) [reply app/hooks/use-account-limits.ts:6]: > Can you move this into galoy-client?

Not at the moment. Till the limits PR on the backend is merged. In my opinion, I'd suggest we hold off adding it to the galoy-client until we finalise the KYC mutations and query to avoid continuously making changes to the galoy client code

File: app/screens/settings-screen/account-screen.tsx
Comment (Extheoisah) [reply app/screens/settings-screen/account-screen.tsx:65]: > What about some type of exclamation icon? e.g. `exclamationcircleo`. You can see the choices here https://oblador.github.io/react-native-vector-icons/.

👍

File: app/hooks/use-account-limits.ts
Comment (UncleSamtoshi) [inline app/hooks/use-account-limits.ts:68]: I think this needs to be wrapped in a useEffect so that on every rerender you dont record the same error.

File: app/hooks/use-account-limits.ts
Comment (UncleSamtoshi) [reply app/hooks/use-account-limits.ts:76]: I'm still having trouble understanding why you cant just use the limits directly. It seems that the state is unnecessarily duplicated  which also results in extra renders.

File: app/hooks/use-account-limits.ts
Comment (UncleSamtoshi) [inline app/hooks/use-account-limits.ts:78]: What is the point in using references vs just return the memoized value?

File: app/hooks/use-account-limits.ts
Comment (UncleSamtoshi) [inline app/hooks/use-account-limits.ts:50]: These types might be null right?

**Review by UncleSamtoshi — CHANGES_REQUESTED** [review]

File: app/hooks/use-account-limits.ts
Comment (Extheoisah) [reply app/hooks/use-account-limits.ts:50]: Yes

File: app/hooks/use-account-limits.ts
Comment (Extheoisah) [reply app/hooks/use-account-limits.ts:78]: I'm pushing a refactor. Its much simpler and I eliminated the use of memoised values

File: app/screens/settings-screen/account-limits-screen.tsx
Comment (UncleSamtoshi) [inline app/screens/settings-screen/account-limits-screen.tsx:124]: You should check with @vindard if there will be any guarantee that they are sorted from shortest to longest duration. Otherwise we will have to find a way to sort it on the frontend.

File: app/screens/settings-screen/account-limits-screen.tsx
Comment (UncleSamtoshi) [inline app/screens/settings-screen/account-limits-screen.tsx:168]: This will also need to handle monthly limit, so it may be best to use a map or function that maps typename to a string representing the duration. As it is written currently, if a new month limit was added, it would display as a week limit.

**Review by UncleSamtoshi — CHANGES_REQUESTED** [review]
I think this is the right idea for the current schema of the API!

File: app/screens/settings-screen/account-limits-screen.tsx
Comment (Extheoisah) [reply app/screens/settings-screen/account-limits-screen.tsx:131]: waiting on a reply from him but it seems so from here src/graphql/types/object/account-limits.ts

Comment (daviroo) [reply app/screens/settings-screen/account-limits-screen.tsx:131]: I think that even if the order is coming back from the API in the right way at the moment we should probably have code in the front end which orders it anyway.  If someone changes the order of an array on the backend they wouldn't expect that to change the rendering on the mobile app.

**Comment by daviroo** [comment]
I'm happy for @UncleSamtoshi to approve once his concerns have been addressed.  I've only left 1 comment regarding ordering.

File: app/screens/settings-screen/account-limits-screen.tsx
Comment (vindard) [reply app/screens/settings-screen/account-limits-screen.tsx:131]: Added this as a consideration under `point 3.` here: https://github.com/GaloyMoney/galoy/pull/1953#issuecomment-1357689726

**Comment by daviroo** [comment]
@UncleSamtoshi I'm going to approve this one since it seems @Extheoisah addressed all the concerns.

**Review by daviroo — APPROVED** [review]

**Pull request merged**
